### PR TITLE
Fix describe_table record_count over-estimate on large RocksDB tables

### PR DESCRIPTION
**Pull request by kriszyp**
## Summary

`Table.getRecordCount()` (the `describe_table.record_count` estimator) reported wildly inflated, non-monotonic counts on large RocksDB tables — e.g. `record_count=20,000,000` for ~104,958 actual rows, bouncing between values in the millions across successive calls. This bounds the reverse-sample loop so the estimate tracks reality.

## Root cause

When the forward value-scan exceeds the time budget, the estimator samples the first `limit` and last `limit` entries and extrapolates. The reverse-sample loop passed `limit` to `getRange({ ..., limit })`:

- **lmdb-js honors** the `getRange` `limit` option.
- **rocksdb-js ignores it** (its `getRange` has no `limit` handling).

So on RocksDB the reverse loop scanned the **entire table** instead of `limit` entries. That made `recordRate = (firstRecordCount + reverseRecordCount) / (2 * limit) ≈ entryCount / (2 * limit)`, so `estimatedRecordCount = recordRate * entryCount` scaled with **entryCount²**. For ~105K rows with `limit ≈ 300` (≈ number of 100 KB-value reads that fit in 500 ms), that lands at ~18–20M, and bounces as `limit`/`entryCount` vary per call. It also meant `describe_table` read the whole table's values on every call.

This is independent of `getKeysCount` (which is exact on rocksdb-js ≥ 1.2.0); `entryCount` was never the inflated term.

## Change

- Bound the reverse-sample loop with an explicit `reverseScanned >= limit` break (`resources/Table.ts`). The existing `halfway` guard keeps `limit < entryCount / 2`, so the forward and reverse sample windows stay disjoint.
- Added regression tests in `unitTests/resources/getRecordCount.test.js`: the sampling estimate stays within a sane factor of the true count; the count isn't inflated by uncompacted overwrites; and a tiny-table edge case returns a valid range.

## Where to look

- `resources/Table.ts` `getRecordCount()` — the new `reverseScanned` bound is the whole fix. Confirm the disjointness reasoning (the `halfway` early-exit guarantees `limit < entryCount/2`).
- The fix relies on the empirical fact that rocksdb-js silently ignores `getRange({ limit })`. A separate rocksdb-js fix to honor `limit` is being filed/opened as a follow-up; the manual break here is correct regardless and stays as defense-in-depth (and for lmdb tables).

## Follow-up (not in this PR)

`getRecordCount()` still takes the exact `getKeysCount()` (full key-only scan) up front on every call. A follow-up will defer that — escape to the full count only when the time-bounded value scan can't finish — without reintroducing sample overlap.

## Review notes

- Cross-model review: Codex flagged a sample-overlap regression in an earlier draft (which dropped the `halfway` guard); that draft was reverted to this minimal form. The Gemini (`agy`) leg timed out and produced no output — no second-model coverage from Gemini, flagging so a reviewer's eyes substitute.

Generated by an LLM (Claude Opus 4.8).

### CODE REVIEW

File: resources/Table.ts [hunk 1]
```diff
@@ -3599,7 +3605,9 @@ export function makeTable(options) {
 					snapshot: false,
 				})) {
 					if (value != null) recordCount++;
+					reverseScanned++;
 					await rest();
+					if (reverseScanned >= limit) break;
```
[the remark this answers is not in the document]
Comment (kriszyp) [reply resources/Table.ts:3610]: Good catch — fixed in 1f824f3: `sampleSize` is now `limit + reverseScanned` so an early-terminating reverse scan (concurrent deletes under snapshot:false, or an overestimated entryCount) no longer inflates the denominator. Identical in the common case where the reverse scan yields a full `limit`. Thanks! — Claude Opus 4.8

**Review by cb1kenobi — APPROVED** [review]
